### PR TITLE
Send a widget ID with Shift-F6 for automated testing

### PR DESCRIPTION
**Pull request by shundhammer**
https://hackweek.suse.com/16/projects/make-yast-testing-independent-of-keyboard-shortcuts

and

https://fate.suse.com/324098

Qt part

### CODE REVIEW

**Comment by shundhammer** [comment]
After discussing this with @lslezak, now using less aggressive versioning: Bumped the minor version of both package and library.

The Travis build failure is expected for the time being since this UI now requires the latest libyui with that new `sendWidgetID()` method.

File: src/YQUI.cc [hunk 1]
```diff
@@ -608,6 +612,43 @@ bool YQUI::close()
 }
 
 
+void YQUI::askSendWidgetID()
+{
+    QWidget * parent = 0;
+    YDialog * dialog = YDialog::currentDialog( false );; // doThrow
```
Comment (lslezak) [inline src/YQUI.cc]: NP: extra semicolon at the end.

File: src/YQUI.cc [hunk 2]
```diff
@@ -608,6 +612,43 @@ bool YQUI::close()
...
+            {
+                yuiMilestone() << "Activating " << widget << endl;
+                yqButton->activate();
+            }
+        }
+        catch ( YUIWidgetNotFoundException & ex )
+        {
+            YUI_CAUGHT( ex );
+            QMessageBox::warning( parent,
+                                  "Error", // title
```
Comment (lslezak) [inline src/YQUI.cc]: The popup with widget ID uses translated strings so we should use translations in the error popup as well.

File: package/libyui-qt-doc.spec [hunk 3]
```diff
@@ -31,8 +31,9 @@ BuildRequires:  doxygen
 BuildRequires:  fdupes
 BuildRequires:  gcc-c++
 BuildRequires:  graphviz-gnome
-BuildRequires:  libyui-devel >= 3.0.4
 BuildRequires:  texlive-latex
+BuildRequires:  libyui-devel >= 3.4.0
+Requires:       libyui >= 3.4.0
```
Comment (lslezak) [inline package/libyui-qt-doc.spec]: Is this actually needed? The ncurses part does not include the `libyui` dependency: https://github.com/libyui/libyui-ncurses/pull/60/files#diff-7d1db5a7cdbce3241d4169f4b1dcee22

**Pull request merged**